thunder/distributed/transforms/fsdp_v2.py

```diff
@@ -57,12 +65,166 @@ def __call__(self, bsym: BoundSymbol) -> VISIT_TYPE:
         return VISIT_TYPE.REPLACE
 
 
-@dataclass(frozen=True)
-class FSDPTraceTransform(Transform):
+# When the user calls fsdp(jitted_module), this function does the following
+# - It transforms the ThunderModule jitted_module, materializing and sharding the parameters as `overrides`
+#   in the ThunderModule.
+# - While doing that, it leaves the original user module alone.
+# - It then registers an early transform (callback that runs before prologue is executed) that transforms the
```

Comment:
is it still an early transform or a vanilla transform?

Reply:
```suggestion
# - It then registers an transform (callback that runs before prologue is executed) that transforms the
```